app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsPresenter.kt

```diff
@@ -320,11 +323,20 @@ class PaymentMethodsPresenter(
       inAppPurchaseInteractor.getPaymentMethods(transaction, fiatValue.amount.toString(),
           fiatValue.currency)
           .map { inAppPurchaseInteractor.mergeAppcoins(it) }
+          .doOnSuccess { updateBalanceDao() }
     } else {
       Single.just(listOf(PaymentMethod.APPC))
     }
   }
 
+  private fun updateBalanceDao() {
```

Comment:
What is this function doing?

Reply:
It's supposed to cache the values in the DB, to avoid making request to the services and so taking more time to load the merged appcoins view. I'll leave a comment in the code for the future